Create CRUD models and Repositories.

ilya-frolov opened the pull request:
Check our Models and Repositories, please.

ilya-frolov committed:
Create CRUD models and Repositories.
Comment spring-boot-starter-jdbc dependency in pom.xml.

ilya-frolov committed:
Change References in Models.

ilya-frolov committed:
Correct column PICPATH name in Game.
Add filling all tables from import.sql.

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/entity/User.java:
```diff
@@ -29,4 +26,17 @@ public User(String userName, String password, String email) {
+            inverseJoinColumns = {@JoinColumn(name = "roles_id")})
+    private List<Role> roles;
+
+    @ManyToMany
+    @JoinTable(name = "sales",
+            joinColumns = {@JoinColumn(name = "users_id")},
+            inverseJoinColumns = {@JoinColumn(name = "games_id")})
```
@ManyToMany(mappedBy = "users")

ilya-frolov replied:
Совсем изменил эту зависимость, т. к. в таблице "sales" нам необходимо дополнительное поле.

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/entity/Role.java:
```diff
@@ -0,0 +1,31 @@
+        this.role = role;
+    }
+
+    @ManyToMany
+    @JoinTable(name = "users_roles",
+            joinColumns = {@JoinColumn(name = "roles_id")},
+            inverseJoinColumns = {@JoinColumn(name = "users_id")})
```
@manytomany(mappedBy = "roles")

RomanKashenok replied:
замени на это и будет хорошо

ilya-frolov replied:
Почитал, понял, изменил.

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/entity/Category.java:
```diff
@@ -0,0 +1,29 @@
+        this.name = name;
+    }
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "category")
```
Кажись вы хотели сделать что бы одна игра была в разных категориях, не?

ilya-frolov replied:
Исправил тип зависимости

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/entity/Game.java:
```diff
@@ -0,0 +1,45 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    @Column(name = "GAMENAME")
```
давай всё с маленькой буквы

ilya-frolov replied:
Сделано

RomanKashenok reviewed (changes requested):
Есть над чем часок посидеть
вот тут посмотри про ManyToMany:
https://www.baeldung.com/hibernate-many-to-many

ilya-frolov committed:
1. Add filling all tables from import.sql.
2. Change relationship between Entities User and Game by adding additional column to the unity table Sales and using one-to-many instead of many-to-many. Create class Sale.
3. Correct many-to-many relationships in User, Role, Game classes.

ilya-frolov committed:
Create Repos for all Entities.

iryna-pantsialeyeva committed:
add hibernate initialisation to dev.properties
update schema.sql
update data.sql

ilya-frolov committed:
Change data.sql.
